week3/src/App.jsx

```diff
@@ -0,0 +1,149 @@
+import styled, { ThemeProvider } from 'styled-components'
+import './style/App.css';
+import theme from './style/theme';
+import { useEffect, useState } from 'react';
+import { CARDS } from './cardData.jsx';
+
+function App() {
+  const [ success, setSuccess ] = useState(false);
+  const [ score, setScore ] = useState(0);
+
+  //filpedId의 값이 각 사진의 id == 배열에 id가 있으면 뒤집힌거 순서대로!!
+  const [ filpedId, setFilpedId ] = useState([]);
+  const [ prevOpenedId, setPrevOpenedId ] = useState(null);
+  
+  const [shuffledCards,setShuffledCards ] = useState([]);
+  useEffect(() => 
+    setShuffledCards([...CARDS].sort(()=> Math.random() - 0.5))
+    ,[success])
+
+  function handleFilp(item){
+    //여기에 item이 아니라 event로 보내서 했는데 그러니까 반영이 안되지!!!
+    //useState상태변경을 위해선 item자체를 보내야함!!!
+
+    //만약 filped 길이가 홀수개면 -> 하나 열린거 있느거임
+    //그럼 지금 열린게 같은 sort인지 확인후 추가
+    //만약 열린게 같은 sort가 아니라면 이 전거 pop,이번꺼 추가 안함
+    
+    //만약 짝수개라면 그냥 열기
+    //만약 길이가 13이라면 축하합니다 !
+
+    //전에 오픈했던데 틀렸다면 둘다 닫혀야 함
+    if(prevOpenedId !== null){
+      setFilpedId(
+        filpedId.filter( each => 
+          (each.sort !== prevOpenedId.sort)
+            &&(each !== filpedId[ filpedId.length -2 ])  )
+      )
```

Comment:
p5: 이전의 상태인 flipedId를 활용하셨군요! flipedId라는 상태를 그냥 가져와서 쓰게되면 상태가 자주 바뀌게 되는 경우에는 바뀐 상태가 바로 반영이 안될 수도 있어요! 이런 경우에는 useState의 setter 함수에서 제공하는 이전 상태를 가져와서 사용하면 항상 최신 상태를 반영해서 안정적인 코드를 만들 수 있습니다!

```suggestion
   setFilpedId( (prev) =>
        prev.filter( each => 
          (each.sort !== prevOpenedId.sort)
            &&(each !== filpedId[ filpedId.length -2 ])  )
      )
```

Reply:
오 setter함수에서 제공하는 이전 상태가 있는 줄 몰랐네요 ㅎㅎ 😭 감사해요! 최신상태의 빠른 반영을 위해 수정했습니다.  수정했더니 원래 두장 뒤집고 한번 클릭해야 다시 뒤집히던게 이제 바로 뒤집히네요! 그전상태를 바로 가져와서 그런 것 같아요:)  감사해요!